drivers/ethernet/eth_enc28j60.c

```diff
@@ -600,13 +601,13 @@ static int eth_enc28j60_rx(struct device *dev, uint16_t *vlan_tag)
 		} while (frm_len > 0);
 
 		/* Let's pop the useless CRC */
-		eth_enc28j60_read_mem(dev, NULL, 4);
```

Comment:
>Replaces NULL buffers as some SPI drivers will fail.

Why not fix the driver then? NULL is [valid](lhttps://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/spi.h#L171), @agansari please revert this changes and open a bug report regarding SPI drivers.

Reply:
@jfischer-phytec-iot there are hal devices there are asserts that verify the return buffer is not NULL, this would mean moving the discard dummy buffer from one place (enc28j60) to another (spi drivers such as spi_mcux_flexcomm) and not all spi drivers were tested with NULL buffers.

Reply:
> there are hal devices there are asserts that verify the return buffer is not NULL

That would not happen with a native driver.

> this would mean moving the discard dummy buffer from one place (enc28j60) to another (spi drivers such as spi_mcux_flexcomm)

sounds like "fixing a bug".  would you like to open a bug report for spi_mcux_flexcomm or provide a patch?

>and not all spi drivers were tested with NULL buffers.

But it is not enc28j60 (or any driver that uses SPI) fault. I cannot follow this approach. Why not fix the bug/issue but the symptoms?

Reply:
I agree with @jfischer-phytec-iot here, we should allow NULL here. If HAL is buggy, the fix should be added there. Sorry, I merged this PR too fast.